diff --git b/libcxx/src/orig.CMakeLists.txt a/libcxx/src/CMakeLists.txt
index 31cd243..3a30da9 100644
--- b/libcxx/src/orig.CMakeLists.txt
+++ a/libcxx/src/CMakeLists.txt
@@ -397,6 +397,7 @@ if (LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY)
     experimental/memory_resource.cpp
     )
   add_library(cxx_experimental STATIC ${LIBCXX_EXPERIMENTAL_SOURCES})
+  target_compile_options(cxx_experimental PRIVATE "-fno-lto")
   if (LIBCXX_ENABLE_SHARED)
     target_link_libraries(cxx_experimental cxx_shared)
   else()
