diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bf99441..a92a7b29 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,7 @@ if(WIN32)
 endif()
 
 # X11 stuff. It's in a if() so that we can say REQUIRED
-if(UNIX AND NOT APPLE)
+if(UNIX)
   find_package(X11 REQUIRED)
 endif()
 
@@ -289,9 +289,9 @@ if(WIN32)
   add_subdirectory(win)
 else()
   # No interest in building x related parts on Apple
-  if(NOT APPLE)
+#   if(NOT APPLE)
     add_subdirectory(unix)
-  endif()
+#   endif()
 endif()
 
 if(ENABLE_NLS)
