diff --git a/release/makemacapp.in b/release/makemacapp.in
index 2bfd869e..44e7797b 100644
--- a/release/makemacapp.in
+++ b/release/makemacapp.in
@@ -42,8 +42,7 @@ if [ -f $PACKAGE_NAME.dmg ]; then
 fi
 
 umask 022
-TMPDIR=`mktemp -d /tmp/$PACKAGE_NAME-build.XXXXXX`
-APPROOT="$TMPDIR/dmg/TigerVNC Viewer $VERSION.app"
+APPROOT="./TigerVNC Viewer.app"
 mkdir -p "$APPROOT/Contents/MacOS"
 mkdir -p "$APPROOT/Contents/Resources"
 
@@ -67,6 +66,8 @@ fi
 install -m 644 $SRCDIR/release/tigervnc.icns "$APPROOT/Contents/Resources/"
 install -m 644 release/Info.plist "$APPROOT/Contents/"
 
+exit
+
 install -m 644 $SRCDIR/LICENCE.TXT $TMPDIR/dmg/
 install -m 644 $SRCDIR/README.rst $TMPDIR/dmg/
 
