--- ./src/sdk/orig.editormanager.cpp	2016-05-19 00:20:45.524020000 +0200
+++ ./src/sdk/editormanager.cpp	2017-11-22 20:33:48.654402530 +0100
@@ -1113,7 +1113,7 @@
 
     wxString cmds = cfg->Read(_T("open_containing_folder"), defCmds) + _T(" ");
     const wxString& fullPath = ed->GetFilename();
-#if defined __WXMSW__ || defined __WXMAC__
+#if defined __WXMSW__ || defined __WXMAC__ || defined __WXQT__
     cmds << fullPath;   // Open folder with the file selected
 #else
     if (cmds.StartsWith(defCmds))
--- ./src/sdk/wxscintilla/src/orig.wxscintilla.cpp	2016-02-06 15:26:58.801436000 +0100
+++ ./src/sdk/wxscintilla/src/wxscintilla.cpp	2017-11-22 20:35:12.814098795 +0100
@@ -153,7 +153,7 @@
     EVT_LEFT_DCLICK             (wxScintilla::OnMouseLeftDown)
     EVT_MOTION                  (wxScintilla::OnMouseMove)
     EVT_LEFT_UP                 (wxScintilla::OnMouseLeftUp)
-#if defined(__WXGTK__) || defined(__WXMAC__)
+#if defined(__WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
     EVT_RIGHT_UP                (wxScintilla::OnMouseRightUp)
 #else
     EVT_CONTEXT_MENU            (wxScintilla::OnContextMenu)
--- ./src/plugins/contrib/wxSmithSTC/stc/orig.stc.cpp	2012-04-28 12:01:45.271368000 +0200
+++ ./src/plugins/contrib/wxSmithSTC/stc/stc.cpp	2017-11-22 20:32:48.334736211 +0100
@@ -110,7 +110,7 @@
     EVT_LEFT_DCLICK             (wxStyledTextCtrl::OnMouseLeftDown)
     EVT_MOTION                  (wxStyledTextCtrl::OnMouseMove)
     EVT_LEFT_UP                 (wxStyledTextCtrl::OnMouseLeftUp)
-#if defined(__WXGTK__) || defined(__WXMAC__)
+#if defined(__WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
     EVT_RIGHT_UP                (wxStyledTextCtrl::OnMouseRightUp)
 #else
     EVT_CONTEXT_MENU            (wxStyledTextCtrl::OnContextMenu)
--- ./src/plugins/contrib/wxSmithDemo/orig.wxSmithDemoMain.cpp	2012-08-31 23:39:22.589685000 +0200
+++ ./src/plugins/contrib/wxSmithDemo/wxSmithDemoMain.cpp	2017-11-22 20:32:19.865117869 +0100
@@ -52,7 +52,7 @@
 }
 
 // The application icon.
-#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__)
+#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__) || defined(__WXQT__)
     #include "media/wxsmith.xpm"
 #endif
 
--- src/plugins/contrib/dragscroll/orig.dragscroll.cpp	2016-02-06 15:29:31.423286000 +0100
+++ src/plugins/contrib/dragscroll/dragscroll.cpp	2017-11-24 14:05:15.110083579 +0100
@@ -1633,12 +1633,14 @@
 //      __WXGTK__ MOUSE SCROLLING __WXGTK__
 // ----------------------------------------------------------------------------
 ///////////////////////////////////////////////////////////////////////////////
-#if defined(__WXGTK__) || defined(__WXMAC__)
+#if defined(__WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
 // ----------------------------------------------------------------------------
 void MouseEventsHandler::OnMouseEvent(wxMouseEvent& event)    //GTK
 // ----------------------------------------------------------------------------
 {
 
+// for now:
+#ifndef __WXQT__
     // For efficiency, skip wheel events now
     if ( event.GetEventType() ==  wxEVT_MOUSEWHEEL)
         { event.Skip(); return; }
@@ -1876,6 +1878,7 @@
 
     }//else if (event.Dragging() && m_dragMode != DRAG_NONE)
 
+#endif // __WXQT__
     // pass on the event
     event.Skip();
 
--- ./src/plugins/contrib/FileManager/orig.directorymonitor.cpp	2016-02-13 17:01:49.283357000 +0100
+++ ./src/plugins/contrib/FileManager/directorymonitor.cpp	2017-11-22 20:21:20.199277733 +0100
@@ -22,7 +22,7 @@
     m_info_uri=wxString(c.m_info_uri.c_str());
 }
 
-#if defined(__WXGTK__) || defined(__WXMAC__)
+#if defined(__WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
 
 #include <map>
 
--- ./src/plugins/contrib/codesnippets/editor/orig.dragscroll.cpp	2016-02-06 15:29:31.423286000 +0100
+++ ./src/plugins/contrib/codesnippets/editor/dragscroll.cpp	2017-11-22 20:24:54.091462330 +0100
@@ -1588,7 +1588,7 @@
 //      __WXGTK__ MOUSE SCROLLING __WXGTK__
 // ----------------------------------------------------------------------------
 ///////////////////////////////////////////////////////////////////////////////
-#if defined(__WXGTK__) || defined(__WXMAC__)
+#if defined(__WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
 // ----------------------------------------------------------------------------
 void MouseEventsHandler::OnMouseEvent(wxMouseEvent& event)    //GTK
 // ----------------------------------------------------------------------------
--- ./src/plugins/contrib/keybinder/orig.menuutils.cpp	2016-02-06 15:29:31.423286000 +0100
+++ ./src/plugins/contrib/keybinder/menuutils.cpp	2017-11-22 20:26:15.661829504 +0100
@@ -299,7 +299,7 @@
 // wxMenuCmd
 // ----------------------------------------------------------------------------
 
-#if defined( __WXGTK__) || defined(__WXMAC__)
+#if defined( __WXGTK__) || defined(__WXMAC__) || defined(__WXQT__)
 // ----------------------------------------------------------------------------
 void wxMenuCmd::Update(wxMenuItem* pSpecificMenuItem) //for __WXGTK__
 // ----------------------------------------------------------------------------
--- ./src/plugins/contrib/SpellChecker/wxspellchecker/samples/wxWidgets/orig.wxSpellCheckerDemo.cpp	2011-08-05 11:48:57.115768000 +0200
+++ ./src/plugins/contrib/SpellChecker/wxspellchecker/samples/wxWidgets/wxSpellCheckerDemo.cpp	2017-11-22 20:22:45.489117633 +0100
@@ -27,7 +27,7 @@
 // ----------------------------------------------------------------------------
 
 // the application icon (under Windows and OS/2 it is in resources)
-#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__)
+#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__) || defined(__WXQT__)
     #include "mondrian.xpm"
 #endif
 
--- ./src/src/orig.app.h	2015-07-26 19:08:46.368079000 +0200
+++ ./src/src/app.h	2017-11-22 20:36:07.431040327 +0100
@@ -22,7 +22,7 @@
     #include <wx/docview.h> // recent files history
 #endif
 
-#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__)
+#if defined(__WXGTK__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__) || defined(__WXX11__) || defined(__WXQT__)
     #include "resources/icons/app.xpm"
 #endif
 
--- orig.configure.ac	2017-05-11 01:33:01.814773000 +0200
+++ configure.ac	2017-11-23 21:09:37.267459906 +0100
@@ -259,6 +259,68 @@
 			WX_X11_LIBS="$WX_X11_LIBS -lX11 $X_EXTRA_LIBS"
 		fi
     ;;
+    *wx_qt*)
+        AC_MSG_RESULT(wxQT)
+		dnl
+		dnl pkg-config no longer needed by debugger plugin
+		dnl but is needed by others
+		dnl
+		dnl only do this if it is really needed
+		dnl AM_CONDITIONAL adds _TRUE and _FALSE to the appropriate variables
+		dnl if the condition is true, the *_TRUE variable is empty (!)
+		if test "x$BUILD_FILEMANAGER_TRUE" = "x" -o \
+				"x$BUILD_SPELLCHECKER_TRUE" = "x" -o \
+				"x$BUILD_MOUSESAP_TRUE" = "x" -o \
+				"x$BUILD_EXPORTER_TRUE" = "x" -o \
+				"x$GTK_NOTEBOOK_TRUE" = "x" -o \
+				"x$LINUX_TRUE" = "x" ; then
+
+			if test "x$LINUX_TRUE" = "x" ; then
+				if test "x$HAVE_PKG_CONFIG" = "xyes"; then
+					PKG_CHECK_MODULES([GLIB2], [glib-2.0])
+				fi
+			fi
+
+			if test "x$BUILD_SPELLCHECKER_TRUE" = "x" ; then
+				if test "x$HAVE_PKG_CONFIG" = "xyes"; then
+					PKG_CHECK_MODULES([HUNSPELL], [hunspell])
+				fi
+			fi
+
+			if test "x$BUILD_MOUSESAP_TRUE" = "x"  ; then
+				if test "x$HAVE_PKG_CONFIG" = "xyes"; then
+                    PKG_CHECK_MODULES([QT5], [Qt5OpenGL])
+				fi
+			fi
+
+			if test "x$BUILD_FILEMANAGER_TRUE" = "x" ; then
+				if test "x$HAVE_PKG_CONFIG" = "xyes"; then
+					PKG_CHECK_MODULES([GAMIN], [gamin])
+				fi
+			fi
+
+			if test "x$BUILD_EXPORTER_TRUE" = "x" ; then
+				if test "x$HAVE_PKG_CONFIG" = "xyes"; then
+					PKG_CHECK_MODULES([FONTCONFIG], [fontconfig])
+				fi
+			fi
+
+    	fi
+
+        dnl Check whether boost is installed
+        AX_BOOST_BASE
+
+		CB_GLIB2_CFLAGS="$GLIB2_CFLAGS"
+		CB_GLIB2_LIBS="$GLIB2_LIBS"
+		CB_GAMIN_CFLAGS="$GAMIN_CFLAGS"
+		CB_GAMIN_LIBS="$GAMIN_LIBS"
+		CB_HUNSPELL_CFLAGS="$HUNSPELL_CFLAGS"
+		CB_HUNSPELL_LIBS="$HUNSPELL_LIBS"
+		CB_FONTCONFIG_CFLAGS="$FONTCONFIG_CFLAGS"
+		CB_FONTCONFIG_LIBS="$FONTCONFIG_LIBS"
+		WX_QT_CFLAGS="$QT_CFLAGS"
+		WX_QT_LIBS="$QT_LIBS"
+    ;;
     *wx_msw*)
         AC_MSG_RESULT(wxMSW)
     ;;
