--- src/luaconf.h.orig	2016-12-22 14:08:50.000000000 +0100
+++ src/luaconf.h	2018-05-11 10:34:13.480989999 +0200
@@ -200,7 +200,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
@@ -275,12 +275,12 @@
 ** give a warning about it. To avoid these warnings, change to the
 ** default definition.
 */
-#if defined(__GNUC__) && ((__GNUC__*100 + __GNUC_MINOR__) >= 302) && \
-    defined(__ELF__)		/* { */
-#define LUAI_FUNC	__attribute__((visibility("hidden"))) extern
-#else				/* }{ */
+// #if defined(__GNUC__) && ((__GNUC__*100 + __GNUC_MINOR__) >= 302) && \
+//     defined(__ELF__)		/* { */
+// #define LUAI_FUNC	__attribute__((visibility("hidden"))) extern
+// #else				/* }{ */
 #define LUAI_FUNC	extern
-#endif				/* } */
+// #endif				/* } */
 
 #define LUAI_DDEC	LUAI_FUNC
 #define LUAI_DDEF	/* empty */
