diff --git b/src/loader/orig.loader.c a/src/loader/loader.c
index d64bc7c..48083ab 100644
--- b/src/loader/orig.loader.c
+++ a/src/loader/loader.c
@@ -83,18 +83,32 @@ int
 loader_open_device(const char *device_name)
 {
    int fd;
+   struct stat dum;
+   if (stat(device_name, &dum) == -1 && errno == ENOENT) {
+//       log_(_LOADER_WARNING, "loader_open_device() pid=%d %s doesn't exist\n", getpid(), device_name);
+	 return -1;
+   }
 #ifdef O_CLOEXEC
+   errno = 0;
    fd = open(device_name, O_RDWR | O_CLOEXEC);
    if (fd == -1 && errno == EINVAL)
 #endif
    {
+      errno = 0;
       fd = open(device_name, O_RDWR);
       if (fd != -1)
          fcntl(fd, F_SETFD, fcntl(fd, F_GETFD) | FD_CLOEXEC);
    }
    if (fd == -1 && errno == EACCES) {
-      log_(_LOADER_WARNING, "failed to open %s: %s\n",
-           device_name, strerror(errno));
+	 char *error = strdup(strerror(errno));
+	 char *cwd = getcwd(NULL,0);
+	 if (!cwd) {
+		 cwd = strdup(strerror(errno));
+	 }
+      log_(_LOADER_WARNING, "loader_open_device() pid=%d (cwd=%s) failed to open %s: %s\n",
+           getpid(), cwd, device_name, error);
+	 if (error) free(error);
+	 if (cwd) free(cwd);
    }
    return fd;
 }
