diff --git orig.configure.ac configure.ac
index 71108fc..6eeb462 100644
--- orig.configure.ac
+++ configure.ac
@@ -2,7 +2,7 @@ dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.54)
 AC_INIT([libdca],[0.0.7])
 AC_CONFIG_SRCDIR([src/dcadec.c])
-AC_CONFIG_AUX_DIR(.auto)
+dnl AC_CONFIG_AUX_DIR(.auto)
 AC_CONFIG_FILES([Makefile include/Makefile test/Makefile
     src/Makefile libdca/Makefile libao/Makefile vc++/Makefile
     libdca/libdca.pc libdca/libdts.pc])
