diff --git src/plugins/coreplugin/editormanager/editormanager.cpp src/plugins/coreplugin/editormanager/editormanager.cpp
index afb4404878d..e1675bc7327 100644
--- src/plugins/coreplugin/editormanager/editormanager.cpp
+++ src/plugins/coreplugin/editormanager/editormanager.cpp
@@ -1925,17 +1925,21 @@ void EditorManagerPrivate::updateWindowTitleForDocument(IDocument *document, QWi
     const QString dashSep(" - ");
 
     const QString documentName = document ? document->displayName() : QString();
-    if (!documentName.isEmpty())
+	if (!documentName.isEmpty()) {
         windowTitle.append(documentName);
+		if (document && document->isModified()) {
+			windowTitle.append(" [*]");
+		}
+	}
 
     const QString filePath = document ? document->filePath().toFileInfo().absoluteFilePath()
                               : QString();
     const QString windowTitleAddition = d->m_titleAdditionHandler
             ? d->m_titleAdditionHandler(filePath)
-            : QString();
-    if (!windowTitleAddition.isEmpty()) {
-        if (!windowTitle.isEmpty())
-            windowTitle.append(" ");
+			: QString();
+	if (!windowTitleAddition.isEmpty()) {
+		if (!windowTitle.isEmpty())
+			windowTitle.append(" ");
         windowTitle.append(windowTitleAddition);
     }
 
@@ -1963,12 +1967,12 @@ void EditorManagerPrivate::updateWindowTitleForDocument(IDocument *document, QWi
     window->window()->setWindowTitle(windowTitle);
     window->window()->setWindowFilePath(filePath);
 
-    if (HostOsInfo::isMacHost()) {
+//    if (HostOsInfo::isMacHost()) {
         if (document)
             window->window()->setWindowModified(document->isModified());
         else
             window->window()->setWindowModified(false);
-    }
+//    }
 }
 
 void EditorManagerPrivate::updateWindowTitle()
