diff --git src/plugins/coreplugin/actionmanager/actionmanager.cpp src/plugins/coreplugin/actionmanager/actionmanager.cpp
index 391cc077f41..1e7222f11db 100644
--- src/plugins/coreplugin/actionmanager/actionmanager.cpp
+++ src/plugins/coreplugin/actionmanager/actionmanager.cpp
@@ -138,8 +138,7 @@ ActionManager::ActionManager(QObject *parent)
 {
     m_instance = this;
     d = new ActionManagerPrivate;
-    if (Utils::HostOsInfo::isMacHost())
-        QCoreApplication::setAttribute(Qt::AA_DontShowIconsInMenus);
+   QCoreApplication::setAttribute(Qt::AA_DontShowIconsInMenus);
 }
 
 /*!
