diff --git src/pipewire/orig.mem.c src/pipewire/mem.c
index 5a2bcc8..06ad0bf 100644
--- src/pipewire/orig.mem.c
+++ src/pipewire/mem.c
@@ -35,6 +35,37 @@ PW_LOG_TOPIC_EXTERN(log_mem);
  * a whole conflicts with the original glibc header <fnctl.h>.
  */
 
+#ifndef SYS_memfd_create
+#  ifndef __NR_memfd_create
+#    if defined __x86_64__
+#      define __NR_memfd_create 319
+#    elif defined __arm__
+#      define __NR_memfd_create 385
+#    elif defined __aarch64__
+#      define __NR_memfd_create 279
+#    elif defined __s390__
+#      define __NR_memfd_create 350
+#    elif defined _MIPS_SIM
+#      if _MIPS_SIM == _MIPS_SIM_ABI32
+#        define __NR_memfd_create 4354
+#      endif
+#      if _MIPS_SIM == _MIPS_SIM_NABI32
+#        define __NR_memfd_create 6318
+#      endif
+#      if _MIPS_SIM == _MIPS_SIM_ABI64
+#        define __NR_memfd_create 5314
+#      endif
+#    elif defined __i386__
+#      define __NR_memfd_create 356
+#    elif defined __arc__
+#      define __NR_memfd_create 279
+#    else
+#      warning "__NR_memfd_create unknown for your architecture"
+#    endif
+#  endif
+#define SYS_memfd_create __NR_memfd_create
+#endif
+
 static inline int memfd_create(const char *name, unsigned int flags)
 {
 	return syscall(SYS_memfd_create, name, flags);
