diff --git gdata/orig.meson.build gdata/meson.build
index a42bae7..9478d38 100644
--- gdata/orig.meson.build
+++ gdata/meson.build
@@ -134,6 +134,11 @@ enum_headers += custom_target(
 symbol_map = meson.current_source_dir() / 'symbol.map'
 ldflags = cc.get_supported_link_arguments('-Wl,--version-script,' + symbol_map)
 
+compat_version = '@0@.@1@.@2@'.format(
+  26, 0, 0)
+current_version = '@0@.@1@.@2@'.format(
+  current - age,
+  0, 0)
 libgdata_lib = shared_library(
   'gdata',
   include_directories: incs,
@@ -144,6 +149,7 @@ libgdata_lib = shared_library(
   link_depends: symbol_map,
   install: true,
   version: gdata_soversion,
+  darwin_versions: [compat_version, current_version],
   gnu_symbol_visibility: 'default',
 )
 
