diff --git a/lib-src/FileDialog/FileDialog.h b/lib-src/FileDialog/FileDialog.h
index acd57cff8176928856007a7e90972481ee0b7e47..89b3829ef114bd23167a8d943d25e951ed4fc0d5 100644
--- a/lib-src/FileDialog/FileDialog.h
+++ b/lib-src/FileDialog/FileDialog.h
@@ -49,6 +49,8 @@ protected:
 #include "mac/FileDialogPrivate.h"
 #elif defined(__WXMSW__)
 #include "win/FileDialogPrivate.h"
+#elif defined(__WXQT__)
+using FileDialog = wxFileDialog;
 #else
 #error Unknown implementation
 #endif
diff --git a/lib-src/FileDialog/gtk/FileDialogPrivate.cpp b/lib-src/FileDialog/gtk/FileDialogPrivate.cpp
index 0cff20b443461cc1279ea889a6360d77bac91ff4..70078abafd038cbcb84a02ff8d5dd39de19f757a 100644
--- a/lib-src/FileDialog/gtk/FileDialogPrivate.cpp
+++ b/lib-src/FileDialog/gtk/FileDialogPrivate.cpp
@@ -9,6 +9,8 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
+#ifdef __WXGTK__
+
 #include <gtk/gtk.h>
 
 #include "../FileDialog.h"
@@ -612,3 +614,4 @@ void FileDialog::GTKFilterChanged()
     GetEventHandler()->ProcessEvent(event);
 }
 
+#endif //__WXGTK__
