diff --git a/src/toolbars/ToolManager.cpp b/src/toolbars/ToolManager.cpp
index a4516a10a..79bc11947 100644
--- a/src/toolbars/ToolManager.cpp
+++ b/src/toolbars/ToolManager.cpp
@@ -55,6 +55,8 @@
 #include "../AColor.h"
 #include "../AllThemeResources.h"
 #include "../ImageManipulation.h"
+// hack to deactivate a bit in wx/confbase.h on which clang chokes:
+#define __VMS
 #include "../Prefs.h"
 #include "../Project.h"
 #include "../widgets/AButton.h"
