diff --git a/src/toolbars/ControlToolBar.cpp b/src/toolbars/ControlToolBar.cpp
index b4c8cf6f6a121b5cfff8d78e5061c01af9d80213..892b180f3cb626e8225c252c85a22645de6acaab 100644
--- a/src/toolbars/ControlToolBar.cpp
+++ b/src/toolbars/ControlToolBar.cpp
@@ -348,6 +348,14 @@ void ControlToolBar::UpdatePrefs()
    ToolBar::UpdatePrefs();
 }
 
+#if defined(USE_AQUA_THEME) && !defined(__WXMAC__)
+// add some additional spacing to distribute the control buttons in
+// the toolbar, which is given an oversized minimal width
+#define ADD_SPACING(dx,dy) mSizer->Add((dx),(dy))
+#else
+#define ADD_SPACING(dx,dy) /**/
+#endif
+
 void ControlToolBar::ArrangeButtons()
 {
    int flags = wxALIGN_CENTER | wxRIGHT;
@@ -363,6 +371,7 @@ void ControlToolBar::ArrangeButtons()
 
    // Start with a little extra space
    mSizer->Add( 5, 55 );
+   ADD_SPACING( 17, 0 );
 
    // Add the buttons in order based on ergonomic setting
    if( mErgonomicTransportButtons )
@@ -374,10 +383,15 @@ void ControlToolBar::ArrangeButtons()
       mFF->MoveBeforeInTabOrder( mRecord );
 
       mSizer->Add( mPause,  0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mPlay,   0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mStop,   0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mRewind, 0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mFF,     0, flags, 10 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mRecord, 0, flags, 5 );
    }
    else
@@ -389,10 +403,15 @@ void ControlToolBar::ArrangeButtons()
       mStop->MoveBeforeInTabOrder( mFF );
 
       mSizer->Add( mRewind, 0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mPlay,   0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mRecord, 0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mPause,  0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mStop,   0, flags, 2 );
+      ADD_SPACING( 7, 0 );
       mSizer->Add( mFF,     0, flags, 5 );
    }
 
