diff --git a/lib-src/lv2/build b/lib-src/lv2/build
index 3ef13a0953bc33b7dde5cb1874b84710feebd788..6f883d5698ece002d181be624436b58fbe1842db 100755
--- a/lib-src/lv2/build
+++ b/lib-src/lv2/build
@@ -16,7 +16,7 @@ function waf
    pkg=$1
    pushd >/dev/null ${pkg}
    shift
-   $(which python python2 | tail -1) waf --prefix="." --include="." $@ build || exit 1
+   @@PYTHON@@ waf --prefix="." --include="." $@ build || exit 1
    popd >/dev/null 
 
    . .buildvars
@@ -34,11 +34,11 @@ function waf
 
 [ -e liblv2.a ] && exit 0
 
-waf lv2    --no-plugins
-waf serd   --static --no-shared --no-utils
-waf sord   --static --no-shared --no-utils
-waf sratom --static --no-shared
-waf lilv   --static --no-shared --no-utils
-waf suil   --static --no-shared --no-qt
+waf lv2    -vvv --no-plugins
+waf serd   -vvv --static --no-shared --no-utils
+waf sord   -vvv --static --no-shared --no-utils
+waf sratom -vvv --static --no-shared
+waf lilv   -vvv --static --no-shared --no-utils
+waf suil   -vvv --static --no-shared --no-qt
 
 exit 0
diff --git a/lib-src/lv2/configure b/lib-src/lv2/configure
index 05ddb59..488883e 100755
--- a/lib-src/lv2/configure
+++ b/lib-src/lv2/configure
@@ -31,7 +31,7 @@ function waf
       fi
    done
 
-   $(which python python2 | tail -1) waf --prefix="." --include="." $@ configure || exit 1
+   @@PYTHON@@ waf --prefix="." --include="." $@ configure || exit 1
    popd >/dev/null
 
    if [ "${pkg}" == "lv2" ]
