diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index edce76f7..ec93a884 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -873,6 +873,7 @@ int main(int argc, char *argv[])
 #elif defined(__WXGTK__) && defined(NDEBUG)
 
 // Specially define main() for Linux debug
+#include <X11/Xlib.h>
 
 IMPLEMENT_APP_NO_MAIN(AudacityApp)
 IMPLEMENT_WX_THEME_SUPPORT
@@ -889,15 +890,17 @@ int main(int argc, char *argv[])
    // never be able to get rid of the messages entirely, but we should
    // look into what's causing them, so allow them to show in Debug
    // builds.
-   freopen("/dev/null", "w", stdout);
-   freopen("/dev/null", "w", stderr);
+//    freopen("/dev/null", "w", stdout);
+//    freopen("/dev/null", "w", stderr);
 
+   XInitThreads();
    return wxEntry(argc, argv);
 }
 
 #elif defined(__WXGTK__)
 
 // Linux release build
+#include <X11/Xlib.h>
 
 wxIMPLEMENT_WX_THEME_SUPPORT
 int main(int argc, char *argv[])
@@ -907,6 +910,7 @@ int main(int argc, char *argv[])
 
    wxDISABLE_DEBUG_SUPPORT();
 
+   XInitThreads();
    return wxEntry(argc, argv);
 }
 wxIMPLEMENT_APP_NO_MAIN(AudacityApp);
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 61b7b541..fd04876e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1129,6 +1129,7 @@ set( LIBRARIES
       $<$<BOOL:${USE_VAMP}>:libvamp>
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GLIB>
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GTK>
+      $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:-lX11>
       $<$<TARGET_EXISTS:Threads::Threads>:Threads::Threads>
 )
 
