diff --git a/lib/widgets/ephy-certificate-dialog.c b/lib/widgets/ephy-certificate-dialog.c
index a8df188..0be1f53 100644
--- a/lib/widgets/ephy-certificate-dialog.c
+++ b/lib/widgets/ephy-certificate-dialog.c
@@ -345,7 +345,7 @@ ephy_certificate_dialog_new (GtkWindow           *parent,
                                      "security-level", security_level,
                                      "tls-errors", tls_errors,
                                      "modal", TRUE,
-                                     "use-header-bar", TRUE,
+                                     "use-header-bar", FALSE,
                                      NULL));
   if (parent)
     gtk_window_set_transient_for (GTK_WINDOW (dialog), parent);
diff --git a/src/bookmarks/ephy-bookmark-row.c b/src/bookmarks/ephy-bookmark-row.c
index 3f5414f..b367f83 100644
--- a/src/bookmarks/ephy-bookmark-row.c
+++ b/src/bookmarks/ephy-bookmark-row.c
@@ -60,7 +60,7 @@ ephy_bookmark_row_button_clicked_cb (EphyBookmarkRow *row,
   dialog = g_object_new (GTK_TYPE_DIALOG,
                          "title", _("Bookmark Properties"),
                          "transient-for", GTK_WINDOW (gtk_widget_get_toplevel (GTK_WIDGET (row))),
-                         "use-header-bar", TRUE,
+                         "use-header-bar", FALSE,
                          "modal", TRUE,
                          NULL);
 
diff --git a/src/cookies-dialog.c b/src/cookies-dialog.c
index 0915708..47ca8ea 100644
--- a/src/cookies-dialog.c
+++ b/src/cookies-dialog.c
@@ -266,6 +266,6 @@ EphyCookiesDialog *
 ephy_cookies_dialog_new (void)
 {
   return g_object_new (EPHY_TYPE_COOKIES_DIALOG,
-                       "use-header-bar", TRUE,
+                       "use-header-bar", FALSE,
                        NULL);
 }
diff --git a/src/ephy-encoding-dialog.c b/src/ephy-encoding-dialog.c
index 1dfa770..d91dc12 100644
--- a/src/ephy-encoding-dialog.c
+++ b/src/ephy-encoding-dialog.c
@@ -521,7 +521,7 @@ EphyEncodingDialog *
 ephy_encoding_dialog_new (EphyWindow *parent)
 {
   return g_object_new (EPHY_TYPE_ENCODING_DIALOG,
-                       "use-header-bar", TRUE,
+                       "use-header-bar", FALSE,
                        "parent-window", parent,
                        NULL);
 }
diff --git a/src/ephy-search-engine-dialog.c b/src/ephy-search-engine-dialog.c
index c8f5a97..9d68c03 100644
--- a/src/ephy-search-engine-dialog.c
+++ b/src/ephy-search-engine-dialog.c
@@ -707,6 +707,6 @@ EphySearchEngineDialog *
 ephy_search_engine_dialog_new (void)
 {
   return g_object_new (EPHY_TYPE_SEARCH_ENGINE_DIALOG,
-                       "use-header-bar", TRUE,
+                       "use-header-bar", FALSE,
                        NULL);
 }
diff --git a/src/ephy-shell.c b/src/ephy-shell.c
index f62d625..d6ccb00 100644
--- a/src/ephy-shell.c
+++ b/src/ephy-shell.c
@@ -1000,7 +1000,7 @@ ephy_shell_get_prefs_dialog (EphyShell *shell)
 {
   if (shell->prefs_dialog == NULL) {
     shell->prefs_dialog = g_object_new (EPHY_TYPE_PREFS_DIALOG,
-                                        "use-header-bar", TRUE,
+                                        "use-header-bar", FALSE,
                                         NULL);
     g_signal_connect (shell->prefs_dialog,
                       "destroy",
@@ -1205,9 +1205,9 @@ ephy_shell_open_uris_idle (OpenURIsData *data)
   }
 
   /* Set address from the very beginning. Looks odd in app mode if it appears later on. */
-  header_bar = EPHY_HEADER_BAR (ephy_window_get_header_bar (data->window));
-  title_widget = ephy_header_bar_get_title_widget (header_bar);
-  ephy_title_widget_set_address (title_widget, url);
+//   header_bar = EPHY_HEADER_BAR (ephy_window_get_header_bar (data->window));
+//   title_widget = ephy_header_bar_get_title_widget (header_bar);
+//   ephy_title_widget_set_address (title_widget, url);
 
   data->current_uri++;
   data->previous_embed = embed;
diff --git a/src/passwords-dialog.c b/src/passwords-dialog.c
index a6cc4a1..d0d1c3b 100644
--- a/src/passwords-dialog.c
+++ b/src/passwords-dialog.c
@@ -518,6 +518,6 @@ ephy_passwords_dialog_new (EphyPasswordManager *manager)
 {
   return EPHY_PASSWORDS_DIALOG (g_object_new (EPHY_TYPE_PASSWORDS_DIALOG,
                                               "password-manager", manager,
-                                              "use-header-bar", TRUE,
+                                              "use-header-bar", FALSE,
                                               NULL));
 }
diff --git a/src/prefs-dialog.c b/src/prefs-dialog.c
index dc967fa..0cc47f4 100644
--- a/src/prefs-dialog.c
+++ b/src/prefs-dialog.c
@@ -1664,7 +1664,7 @@ clear_personal_data_button_clicked_cb (GtkWidget   *button,
   ClearDataDialog *clear_dialog;
 
   clear_dialog = g_object_new (EPHY_TYPE_CLEAR_DATA_DIALOG,
-                               "use-header-bar", TRUE,
+                               "use-header-bar", FALSE,
                                NULL);
   gtk_window_set_transient_for (GTK_WINDOW (clear_dialog), GTK_WINDOW (dialog));
   gtk_window_set_modal (GTK_WINDOW (clear_dialog), TRUE);
diff --git a/src/resources/gtk/prefs-dialog.ui b/src/resources/gtk/prefs-dialog.ui
index 1009a56..c3cb8b5 100644
--- a/src/resources/gtk/prefs-dialog.ui
+++ b/src/resources/gtk/prefs-dialog.ui
@@ -10,12 +10,6 @@
     <property name="type-hint">dialog</property>
     <property name="default-width">600</property>
     <property name="default-height">800</property>
-    <child internal-child="headerbar">
-      <object class="GtkHeaderBar">
-        <property name="title" translatable="yes">Preferences</property>
-        <property name="show-close-button">True</property>
-      </object>
-    </child>
     <child internal-child="vbox">
       <object class="GtkBox">
         <property name="visible">True</property>
diff --git a/src/resources/gtk/search-engine-dialog.ui b/src/resources/gtk/search-engine-dialog.ui
index b196973..4ded669 100644
--- a/src/resources/gtk/search-engine-dialog.ui
+++ b/src/resources/gtk/search-engine-dialog.ui
@@ -6,12 +6,6 @@
       <property name="window_position">center</property>
       <property name="destroy_with_parent">True</property>
       <property name="type_hint">dialog</property>
-      <child internal-child="headerbar">
-        <object class="GtkHeaderBar">
-          <property name="title" translatable="yes">Manage Search Engines</property>
-          <property name="show-close-button">True</property>
-        </object>
-      </child>
     <child internal-child="vbox">
       <object class="GtkBox">
         <property name="can_focus">False</property>
diff --git a/src/resources/gtk/synced-tabs-dialog.ui b/src/resources/gtk/synced-tabs-dialog.ui
index c8137e2..d25c524 100644
--- a/src/resources/gtk/synced-tabs-dialog.ui
+++ b/src/resources/gtk/synced-tabs-dialog.ui
@@ -17,12 +17,6 @@
     <property name="window_position">center</property>
     <property name="destroy_with_parent">True</property>
     <property name="type_hint">dialog</property>
-    <child internal-child="headerbar">
-      <object class="GtkHeaderBar">
-        <property name="title" translatable="yes">Synced Tabs</property>
-        <property name="show-close-button">True</property>
-      </object>
-    </child>
     <child internal-child="vbox">
       <object class="GtkBox">
         <property name="visible">True</property>
diff --git a/src/synced-tabs-dialog.c b/src/synced-tabs-dialog.c
index 70034b2..7babcdc 100644
--- a/src/synced-tabs-dialog.c
+++ b/src/synced-tabs-dialog.c
@@ -350,7 +350,7 @@ SyncedTabsDialog *
 synced_tabs_dialog_new (EphyOpenTabsManager *manager)
 {
   return EPHY_SYNCED_TABS_DIALOG (g_object_new (EPHY_TYPE_SYNCED_TABS_DIALOG,
-                                                "use-header-bar", TRUE,
+                                                "use-header-bar", FALSE,
                                                 "open-tabs-manager", manager,
                                                 NULL));
 }
diff --git a/src/webapp-additional-urls-dialog.c b/src/webapp-additional-urls-dialog.c
index b57db25..a18bc96 100644
--- a/src/webapp-additional-urls-dialog.c
+++ b/src/webapp-additional-urls-dialog.c
@@ -264,6 +264,6 @@ EphyWebappAdditionalURLsDialog *
 ephy_webapp_additional_urls_dialog_new (void)
 {
   return EPHY_WEBAPP_ADDITIONAL_URLS_DIALOG (g_object_new (EPHY_TYPE_WEBAPP_ADDITIONAL_URLS_DIALOG,
-                                                           "use-header-bar", TRUE,
+                                                           "use-header-bar", FALSE,
                                                            NULL));
 }
diff --git a/src/window-commands.c b/src/window-commands.c
index 081f7f9..abdadaa 100644
--- a/src/window-commands.c
+++ b/src/window-commands.c
@@ -229,7 +229,7 @@ show_profile_selector (GtkWidget *parent, GSList *profiles)
 
   selector = gtk_dialog_new_with_buttons (_("Select Profile"),
                                           GTK_WINDOW (parent),
-                                          GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT | GTK_DIALOG_USE_HEADER_BAR,
+                                          GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
                                           _("_Cancel"),
                                           GTK_RESPONSE_CANCEL,
                                           _("_Select"),
diff --git a/subprojects/libhandy/src/hdy-dialog.c b/subprojects/libhandy/src/hdy-dialog.c
index a9827ba..993bbe6 100644
--- a/subprojects/libhandy/src/hdy-dialog.c
+++ b/subprojects/libhandy/src/hdy-dialog.c
@@ -375,7 +375,7 @@ GtkWidget *
 hdy_dialog_new (GtkWindow *parent)
 {
   return g_object_new (HDY_TYPE_DIALOG,
-                       "use-header-bar", TRUE,
+                       "use-header-bar", FALSE,
                        "transient-for", parent,
                        NULL);
 }
diff --git a/subprojects/libhandy/src/hdy-header-group.c b/subprojects/libhandy/src/hdy-header-group.c
index 864e476..bbebeb8 100644
--- a/subprojects/libhandy/src/hdy-header-group.c
+++ b/subprojects/libhandy/src/hdy-header-group.c
@@ -255,7 +255,8 @@ hdy_header_group_set_focus (HdyHeaderGroup *self,
 
   priv = hdy_header_group_get_instance_private (self);
 
-  priv->focus = header_bar;
+//   priv->focus = header_bar;
+  priv->focus = NULL;
 
   update_decoration_layouts (self);
 
