diff --git orig.CMakeLists.txt CMakeLists.txt
index e234cb6..91de7e8 100644
--- orig.CMakeLists.txt
+++ CMakeLists.txt
@@ -3,7 +3,11 @@
 cmake_minimum_required(VERSION 3.5)
 
 # Update the version for each new release
-project(fontforge VERSION 20230101 LANGUAGES C CXX)
+if(APPLE)
+  project(fontforge VERSION 20230101 LANGUAGES C CXX OBJC)
+else()
+  project(fontforge VERSION 20230101 LANGUAGES C CXX)
+endif()
 
 # No in source builds
 if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
