diff --git fontforgeexe/orig.startui.c fontforgeexe/startui.c
index 307a442..972d656 100644
--- fontforgeexe/orig.startui.c
+++ fontforgeexe/startui.c
@@ -622,8 +622,13 @@ int fontforge_main( int argc, char **argv ) {
 
 #ifdef __Mac
     extern void setup_cocoa_app(void);
-    setup_cocoa_app();
-    hotkeySystemSetCanUseMacCommand(true);
+    char *backend = getenv("GDK_BACKEND");
+    if (backend && strncmp(backend, "quartz", 6) == 0 ){
+     setup_cocoa_app();
+     hotkeySystemSetCanUseMacCommand(true);
+    } else {
+	putenv("GDK_BACKEND=x11");
+    }
 #endif
 
     /* Must be done before we cache the current directory */
@@ -1023,8 +1028,10 @@ exit( 0 );
 	any = ReopenLastFonts();
 
 #ifdef __Mac
-    extern bool launch_cocoa_app(void);
-    any = launch_cocoa_app() || any;
+    if (backend && strncmp(backend, "quartz", 6) == 0 ){
+     extern bool launch_cocoa_app(void);
+     any = launch_cocoa_app() || any;
+    }
 #endif
 
     if ( doopen || !any )
