--- /opt/local/var/macports/sources/svn.macports.org/trunk/dports/multimedia/ffmpeg/Portfile	2016-06-22 18:02:34.964446067 +0200
+++ Portfile	2016-06-22 19:08:15.054267863 +0200
@@ -44,7 +44,8 @@
                     and high code reusability. \
                     \
                     libavformat is a library containing parsers and \
-                    generators for all common audio/video formats.
+                    generators for all common audio/video formats. \
+				This is a variant adapted for Linux, taking most dependencies from the host.
 
 platforms           darwin
 homepage            http://www.ffmpeg.org/
@@ -55,38 +56,19 @@
 checksums           rmd160  7255df119a4efef1b0da58e6e148028edeeed7f2 \
                     sha256  30e3c77c2f4c358ed087869455a7496cbd7753a5e1b98d20ba49c1004009fd36
 
-depends_build       port:pkgconfig \
-                    port:gmake \
-                    port:texinfo
+depends_build       port:pkgconfig
 
 # libvpx is static only so can be considered a build dependency (#47934)
 
 depends_build-append \
                     port:libvpx
 
-depends_lib         port:lame \
-                    port:libiconv \
-                    port:libvorbis \
-                    port:libopus \
-                    port:libogg \
-                    port:libtheora \
-                    port:libmodplug \
-                    port:schroedinger \
-                    port:libass \
-                    port:libbluray \
-                    port:gnutls \
-                    port:openjpeg \
-                    port:fontconfig \
-                    port:freetype \
-                    port:fribidi \
-                    path:lib/libspeex.dylib:speex \
-                    port:libsdl \
-                    port:soxr \
+depends_lib         port:libiconv \
                     port:bzip2 \
                     port:xz \
                     port:zlib
 
-build.cmd           ${prefix}/bin/gmake
+#build.cmd           ${prefix}/bin/gmake
 build.env-append    V=1
 
 #
@@ -131,8 +113,6 @@
                     --disable-libxcb-xfixes \
                     --disable-indev=jack \
                     --disable-outdev=xv \
-                    --disable-vda \
-                    --disable-videotoolbox \
                     --mandir=${prefix}/share/man \
                     --enable-shared --enable-pthreads \
                     --cc=${configure.cc}
@@ -175,7 +155,7 @@
     }
 }
 
-test.run            yes
+# test.run            yes
 
 #
 # configure isn't autoconf and they do use a dep cache
@@ -211,7 +191,7 @@
         lappend merger_configure_env($arch)  ASFLAGS='-arch ${arch}'
     }
     if {[string match "*86*" ${configure.universal_archs}]} {
-        depends_build-append port:yasm
+        #depends_build-append port:yasm
     }
     lappend merger_configure_args(i386) --enable-yasm
     lappend merger_configure_args(x86_64) --enable-yasm
@@ -219,7 +199,7 @@
     configure.args-append --arch=${configure.build_arch}
     configure.env-append  ASFLAGS='[get_canonical_archflags]'
     if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
-        depends_build-append port:yasm
+        #depends_build-append port:yasm
         configure.args-append --enable-yasm
     }
 }
@@ -228,9 +208,11 @@
 #
 # enable x11grab_xcb input device
 #
+    platform darwin {
     depends_lib-append      port:xorg-libxcb \
                             port:xorg-libXext \
                             port:xorg-libXfixes
+    }
 
     configure.args-delete   --disable-libxcb \
                             --disable-libxcb-shm \
@@ -240,9 +222,9 @@
 default_variants +x11
 
 if {[variant_isset x11]} {
-    require_active_variants libsdl x11
+    #require_active_variants libsdl x11
 } else {
-    require_active_variants libsdl "" x11
+    #require_active_variants libsdl "" x11
 }
 
 variant libdc1394 description {Enable IIDC-1394 frame grabbing using libdc1394 (experimental)} {
@@ -289,8 +271,10 @@
                             --enable-postproc \
                             --enable-libx264 \
                             --enable-libxvid
+    platform darwin {
     depends_lib-append      port:XviD \
                             port:x264
+    }
 
     license                 GPL-2+
 }
@@ -301,12 +285,16 @@
                             --enable-libopencore-amrwb \
                             --enable-libsmbclient
 
+    platform darwin {
     depends_lib-append      port:opencore-amr \
                             port:samba3
+    }
 
     license                 GPL-3+
 }
 
+default_variants-append    +gpl3
+
 # the build server uses the default variants, and we want distributable binaries
 # nonfree code is disabled by default but can be enabled using the +nonfree variant
 
@@ -314,8 +302,10 @@
     configure.args-append   --enable-nonfree \
                             --enable-libfdk-aac \
                             --enable-libfaac
+    platform darwin {
     depends_lib-append      port:libfdk-aac \
                             port:faac
+    }
     license                 Restrictive
 }
 
