diff --git orig.configure.ac configure.ac
index 7b651b6..d923eeb 100644
--- orig.configure.ac
+++ configure.ac
@@ -154,7 +154,6 @@ PKG_CHECK_MODULES(PRIVACY_PANEL, $COMMON_MODULES)
 PKG_CHECK_MODULES(REGION_PANEL, $COMMON_MODULES
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
                   gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION
-                  accountsservice >= $ACCOUNTSSERVICE_REQUIRED_VERSION
                   $IBUS_MODULE)
 PKG_CHECK_MODULES(SCREEN_PANEL, $COMMON_MODULES)
 PKG_CHECK_MODULES(SEARCH_PANEL, $COMMON_MODULES)
diff --git panels/region/orig.cc-region-panel.c panels/region/cc-region-panel.c
index 48c0d7a..93e24e4 100644
--- panels/region/orig.cc-region-panel.c
+++ panels/region/cc-region-panel.c
@@ -45,7 +45,7 @@
 #include "cc-ibus-utils.h"
 #endif
 
-#include <act/act.h>
+// #include <act/act.h>
 
 #define GNOME_DESKTOP_INPUT_SOURCES_DIR "org.gnome.desktop.input-sources"
 #define KEY_INPUT_SOURCES        "sources"
@@ -95,8 +95,8 @@ struct _CcRegionPanelPrivate {
         GtkListBoxRow *formats_row;
         GtkWidget     *formats_label;
 
-        ActUserManager *user_manager;
-        ActUser        *user;
+//         ActUserManager *user_manager;
+//         ActUser        *user;
         GSettings      *locale_settings;
 
         gchar *language;
@@ -133,15 +133,15 @@ cc_region_panel_finalize (GObject *object)
         g_cancellable_cancel (priv->cancellable);
         g_clear_object (&priv->cancellable);
 
-        if (priv->user_manager) {
-                g_signal_handlers_disconnect_by_data (priv->user_manager, self);
-                priv->user_manager = NULL;
-        }
+//         if (priv->user_manager) {
+//                 g_signal_handlers_disconnect_by_data (priv->user_manager, self);
+//                 priv->user_manager = NULL;
+//         }
 
-        if (priv->user) {
-                g_signal_handlers_disconnect_by_data (priv->user, self);
-                priv->user = NULL;
-        }
+//         if (priv->user) {
+//                 g_signal_handlers_disconnect_by_data (priv->user, self);
+//                 priv->user = NULL;
+//         }
 
         g_clear_object (&priv->permission);
         g_clear_object (&priv->localed);
@@ -365,7 +365,7 @@ update_language (CcRegionPanel *self,
         } else {
                 if (g_strcmp0 (language, priv->language) == 0)
                         return;
-                act_user_set_language (priv->user, language);
+//                 act_user_set_language (priv->user, language);
                 if (priv->login_auto_apply)
                         set_system_language (self, language);
                 maybe_notify (self, LC_MESSAGES, language);
@@ -623,8 +623,8 @@ update_language_from_user (CcRegionPanel *self)
 	CcRegionPanelPrivate *priv = self->priv;
         const gchar *language = NULL;
 
-        if (act_user_is_loaded (priv->user))
-                language = act_user_get_language (priv->user);
+//         if (act_user_is_loaded (priv->user))
+//                 language = act_user_get_language (priv->user);
 
         if (language == NULL || *language == '\0')
                 language = setlocale (LC_MESSAGES, NULL);
@@ -640,11 +640,11 @@ setup_language_section (CcRegionPanel *self)
 	CcRegionPanelPrivate *priv = self->priv;
         GtkWidget *widget;
 
-        priv->user = act_user_manager_get_user_by_id (priv->user_manager, getuid ());
-        g_signal_connect_swapped (priv->user, "notify::language",
-                                  G_CALLBACK (update_language_from_user), self);
-        g_signal_connect_swapped (priv->user, "notify::is-loaded",
-                                  G_CALLBACK (update_language_from_user), self);
+//         priv->user = act_user_manager_get_user_by_id (priv->user_manager, getuid ());
+//         g_signal_connect_swapped (priv->user, "notify::language",
+//                                   G_CALLBACK (update_language_from_user), self);
+//         g_signal_connect_swapped (priv->user, "notify::is-loaded",
+//                                   G_CALLBACK (update_language_from_user), self);
 
         priv->locale_settings = g_settings_new (GNOME_SYSTEM_LOCALE_DIR);
         g_signal_connect_swapped (priv->locale_settings, "changed::" KEY_REGION,
@@ -1729,16 +1729,16 @@ set_login_button_visibility (CcRegionPanel *self)
         gboolean has_multiple_users;
         gboolean loaded;
 
-        g_object_get (priv->user_manager, "is-loaded", &loaded, NULL);
-        if (!loaded)
+//         g_object_get (priv->user_manager, "is-loaded", &loaded, NULL);
+//         if (!loaded)
           return;
 
-        g_object_get (priv->user_manager, "has-multiple-users", &has_multiple_users, NULL);
-
-        priv->login_auto_apply = !has_multiple_users && g_permission_get_allowed (priv->permission);
-        gtk_widget_set_visible (priv->login_button, !priv->login_auto_apply);
-
-        g_signal_handlers_disconnect_by_func (priv->user_manager, set_login_button_visibility, self);
+//         g_object_get (priv->user_manager, "has-multiple-users", &has_multiple_users, NULL);
+// 
+//         priv->login_auto_apply = !has_multiple_users && g_permission_get_allowed (priv->permission);
+//         gtk_widget_set_visible (priv->login_button, !priv->login_auto_apply);
+// 
+//         g_signal_handlers_disconnect_by_func (priv->user_manager, set_login_button_visibility, self);
 }
 
 static void
@@ -1779,12 +1779,12 @@ setup_login_button (CcRegionPanel *self)
         g_signal_connect_swapped (priv->login_button, "notify::active",
                                   G_CALLBACK (login_changed), self);
 
-        g_object_get (priv->user_manager, "is-loaded", &loaded, NULL);
-        if (loaded)
-                set_login_button_visibility (self);
-        else
-                g_signal_connect_swapped (priv->user_manager, "notify::is-loaded",
-                                          G_CALLBACK (set_login_button_visibility), self);
+//         g_object_get (priv->user_manager, "is-loaded", &loaded, NULL);
+//         if (loaded)
+//                 set_login_button_visibility (self);
+//         else
+//                 g_signal_connect_swapped (priv->user_manager, "notify::is-loaded",
+//                                           G_CALLBACK (set_login_button_visibility), self);
 }
 
 static void
@@ -1828,7 +1828,7 @@ cc_region_panel_init (CcRegionPanel *self)
 		return;
 	}
 
-        priv->user_manager = act_user_manager_get_default ();
+//         priv->user_manager = act_user_manager_get_default ();
 
         priv->cancellable = g_cancellable_new ();
 
