diff --git orig.configure.ac configure.ac
index 42ddc62..401f9ab 100644
--- orig.configure.ac
+++ configure.ac
@@ -163,12 +149,6 @@ PKG_CHECK_MODULES(SOUND_PANEL, $COMMON_MODULES libxml-2.0
                   libpulse >= $PA_REQUIRED_VERSION
                   libpulse-mainloop-glib >= $PA_REQUIRED_VERSION)
 PKG_CHECK_MODULES(UNIVERSAL_ACCESS_PANEL, $COMMON_MODULES)
-PKG_CHECK_MODULES(USER_ACCOUNTS_PANEL, $COMMON_MODULES
-                  polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
-                  gnome-desktop-3.0
-                  gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED_VERSION
-                  pwquality >= $PWQUALITY_REQUIRED_VERSION
-                  accountsservice >= $ACCOUNTSSERVICE_REQUIRED_VERSION)
 PKG_CHECK_MODULES(SHARING_PANEL, $COMMON_MODULES)
 PKG_CHECK_MODULES(REMOTE_LOGIN_HELPER, glib-2.0 >= $GLIB_REQUIRED_VERSION gio-2.0)
 
diff --git shell/orig.Makefile.am shell/Makefile.am
index 31da9e2..752f521 100644
--- shell/orig.Makefile.am
+++ shell/Makefile.am
@@ -93,7 +93,6 @@ gnome_control_center_LDADD =						\
 	$(top_builddir)/panels/sharing/libsharing.la			\
 	$(top_builddir)/panels/sound/libsound.la			\
 	$(top_builddir)/panels/universal-access/libuniversal-access.la	\
-	$(top_builddir)/panels/user-accounts/libuser-accounts.la	\
 	$(top_builddir)/panels/printers/libprinters.la
 
 if BUILD_WACOM
diff --git shell/orig.cc-panel-loader.c shell/cc-panel-loader.c
index e8faccd..d3beda5 100644
--- shell/orig.cc-panel-loader.c
+++ shell/cc-panel-loader.c
@@ -106,7 +106,9 @@ static struct {
   PANEL_TYPE("sharing",          cc_sharing_panel_get_type      ),
   PANEL_TYPE("sound",            cc_sound_panel_get_type        ),
   PANEL_TYPE("universal-access", cc_ua_panel_get_type           ),
+#ifdef BUILD_USERACCOUNTS
   PANEL_TYPE("user-accounts",    cc_user_panel_get_type         ),
+#endif
 #ifdef BUILD_WACOM
   PANEL_TYPE("wacom",            cc_wacom_panel_get_type        ),
 #endif
diff --git panels/orig.Makefile.am panels/Makefile.am
index 0495e16..913ed57 100644
--- panels/orig.Makefile.am
+++ panels/Makefile.am
@@ -10,7 +10,6 @@ SUBDIRS= \
 	sound \
 	keyboard \
 	universal-access \
-	user-accounts \
 	datetime \
 	search \
 	privacy \
