diff --git panels/printers/orig.pp-printer.c panels/printers/pp-printer.c
index 98474ef..e12f14e 100644
--- panels/printers/orig.pp-printer.c
+++ panels/printers/pp-printer.c
@@ -18,6 +18,8 @@
  *          Marek Kasik <mkasik@redhat.com>
  */
 
+#define _IPP_PRIVATE_STRUCTURES 1
+
 #include "pp-printer.h"
 
 #include "pp-job.h"
@@ -26,6 +28,10 @@
 #define IPP_STATE_IDLE IPP_IDLE
 #endif
 
+#ifndef ippGetState
+#define ippGetState(ipp)      ipp->state
+#endif
+
 typedef struct _PpPrinter        PpPrinter;
 typedef struct _PpPrinterPrivate PpPrinterPrivate;
 
