diff --git a/src/rofiles-fuse/main.c b/src/rofiles-fuse/main.c
index 0c267ea9..27a3fd5e 100644
--- a/src/rofiles-fuse/main.c
+++ b/src/rofiles-fuse/main.c
@@ -37,12 +37,19 @@
 #include <sys/types.h>
 #include <sys/xattr.h>
 #include <unistd.h>
+#include <linux/stat.h>
 
 #include <glib.h>
 
 #include "libglnx.h"
 #include "ostree.h"
 
+static int statx(int dirfd, const char *restrict pathname, int flags,
+                 unsigned int mask, struct statx *restrict statxbuf)
+{
+	return syscall(__NR_statx, dirfd, pathname, flags, mask, statxbuf);
+}
+
 // Global to store our read-write path
 static int basefd = -1;
 /* Whether or not to automatically "copyup" (in overlayfs terms).
