--- setup.py.orig	2023-08-24 19:59:28.000000000 +0200
+++ setup.py	2023-11-06 23:28:45.000000000 +0100
@@ -44,7 +44,7 @@
 TEST_EXTENSIONS = True
 
 # This global variable is used to hold the list of modules to be disabled.
-DISABLED_MODULE_LIST = []
+DISABLED_MODULE_LIST = ['_tkinter', '_gdbm']
 
 
 def get_platform():
@@ -736,9 +736,6 @@
         # Ensure that /usr/local is always used, but the local build
         # directories (i.e. '.' and 'Include') must be first.  See issue
         # 10520.
-        if not CROSS_COMPILING:
-            add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
-            add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
         if CROSS_COMPILING:
             self.add_cross_compiling_paths()
@@ -1443,17 +1440,12 @@
 
     def detect_sqlite(self):
         # The sqlite interface
-        sqlite_setup_debug = False   # verbose debug prints from this script?
+        sqlite_setup_debug = True   # verbose debug prints from this script?
 
         # We hunt for #define SQLITE_VERSION "n.n.n"
         # We need to find >= sqlite version 3.3.9, for sqlite3_prepare_v2
         sqlite_incdir = sqlite_libdir = None
-        sqlite_inc_paths = [ '/usr/include',
-                             '/usr/include/sqlite',
-                             '/usr/include/sqlite3',
-                             '/usr/local/include',
-                             '/usr/local/include/sqlite',
-                             '/usr/local/include/sqlite3',
+        sqlite_inc_paths = [ '@@PREFIX@@/include',
                              ]
         if CROSS_COMPILING:
             sqlite_inc_paths = []
@@ -1467,7 +1459,7 @@
         if MACOS:
             sysroot = macosx_sdk_root()
 
-        for d_ in self.inc_dirs + sqlite_inc_paths:
+        for d_ in sqlite_inc_paths:
             d = d_
             if MACOS and is_macosx_sdk_path(d):
                 d = os.path.join(sysroot, d[1:])
@@ -1778,10 +1770,7 @@
         uuid_h = sysconfig.get_config_var("HAVE_UUID_H")
         uuid_uuid_h = sysconfig.get_config_var("HAVE_UUID_UUID_H")
         if uuid_h or uuid_uuid_h:
-            if sysconfig.get_config_var("HAVE_LIBUUID"):
-                uuid_libs = ["uuid"]
-            else:
-                uuid_libs = []
+            uuid_libs = []
             self.add(Extension('_uuid', ['_uuidmodule.c'],
                                libraries=uuid_libs))
         else:
@@ -1990,8 +1979,8 @@
         # Rather than complicate the code below, detecting and building
         # AquaTk is a separate method. Only one Tkinter will be built on
         # Darwin - either AquaTk, if it is found, or X11 based Tk.
-        if (MACOS and self.detect_tkinter_darwin()):
-            return True
+        self.announce("INFO: Not using Tcl/Tk", 2)
+        return False
 
         # Assume we haven't found any of the libraries or include files
         # The versions with dots are used on Unix, and the versions without
