diff --git src/qt/menu.cpp src/qt/menu.cpp
index a0433d7229c41ccaa89259fd03bc77bf11c317fd..4edf164eee033cbc4f0c26bc76ae541f2e3fbad4 100644
--- src/qt/menu.cpp
+++ src/qt/menu.cpp
@@ -11,6 +11,7 @@
 #include "wx/menu.h"
 #include "wx/qt/private/utils.h"
 #include "wx/qt/private/converter.h"
+#include "wx/stockitem.h"
 
 #include <QtWidgets/QMenu>
 #include <QtWidgets/QMenuBar>
@@ -56,23 +57,54 @@ static void InsertMenuItemAction( const wxMenu *menu, const wxMenuItem *previous
 {
     QMenu *qtMenu = menu->GetHandle();
     QAction *itemAction = item->GetHandle();
-    if ( item->GetKind() == wxITEM_RADIO )
+    switch ( item->GetKind() )
     {
-        // If the previous menu item is a radio item then add this item to the
-        // same action group, otherwise start a new group:
-
-        if ( previousItem != NULL && previousItem->GetKind() == wxITEM_RADIO )
-        {
-            QAction *previousItemAction = previousItem->GetHandle();
-            QActionGroup *previousItemActionGroup = previousItemAction->actionGroup();
-            wxASSERT_MSG( previousItemActionGroup != NULL, "An action group should have been setup" );
-            previousItemActionGroup->addAction( itemAction );
-        }
-        else
+        case wxITEM_RADIO:
+            // If the previous menu item is a radio item then add this item to the
+            // same action group, otherwise start a new group:
+
+            if ( previousItem != NULL && previousItem->GetKind() == wxITEM_RADIO )
+            {
+                QAction *previousItemAction = previousItem->GetHandle();
+                QActionGroup *previousItemActionGroup = previousItemAction->actionGroup();
+                wxASSERT_MSG( previousItemActionGroup != NULL, "An action group should have been setup" );
+                previousItemActionGroup->addAction( itemAction );
+            }
+            else
+            {
+                QActionGroup *actionGroup = new QActionGroup( qtMenu );
+                actionGroup->addAction( itemAction );
+                wxASSERT_MSG( itemAction->actionGroup() == actionGroup, "Must be the same action group" );
+            }
+            break;
+        case wxITEM_NORMAL:
         {
-            QActionGroup *actionGroup = new QActionGroup( qtMenu );
-            actionGroup->addAction( itemAction );
-            wxASSERT_MSG( itemAction->actionGroup() == actionGroup, "Must be the same action group" );
+            wxWindowID id = item->GetId();
+            if ( wxIsStockID( id ) )
+            {
+                itemAction->setText( wxQtConvertString( wxGetStockLabel( id ) ) );
+                wxAcceleratorEntry accel = wxGetStockAccelerator( id );
+                QString opcode(QStringLiteral("Ctrl+"));
+                // keep this in sync with the flags set in wxGetStockAccelerator()!
+                if ( accel.GetFlags() & wxACCEL_SHIFT )
+                {
+                    opcode += QStringLiteral("Shift+");
+                }
+                QString shortcut;
+                if ( id == wxID_EXIT )
+                {
+                    shortcut = QStringLiteral("Q");
+                }
+                else if ( accel.IsOk() )
+                {
+                    shortcut = wxQtConvertString( accel.ToRawString() );
+                }
+                if ( !shortcut.isEmpty() )
+                {
+                    itemAction->setShortcut( QKeySequence( opcode + shortcut ) );
+                }
+            }
+            break;
         }
     }
     // Insert the action into the actual menu:
@@ -174,7 +206,7 @@ bool wxMenuBar::Append( wxMenu *menu, const wxString& title )
 
     QMenu *qtMenu = SetTitle( menu, title );
     m_qtMenuBar->addMenu( qtMenu );
-    
+
     return true;
 }
 
