--- orig.meson.build	2017-07-19 04:49:23.000000000 +0200
+++ meson.build	2017-08-04 20:00:34.163682993 +0200
@@ -79,12 +79,14 @@
 	   'udev/libinput-device-group.c',
 	   dependencies : [dep_udev, dep_libwacom],
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true,
 	   install_dir : udev_dir)
 executable('libinput-model-quirks',
 	   'udev/libinput-model-quirks.c',
 	   dependencies : dep_udev,
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true,
 	   install_dir : udev_dir)
 
@@ -368,6 +370,7 @@
 	   libinput_debug_events_sources,
 	   dependencies : deps_tools,
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true
 	   )
@@ -383,6 +386,7 @@
 	   libinput_list_devices_sources,
 	   dependencies : deps_tools,
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -398,6 +402,7 @@
 	   libinput_measure_sources,
 	   dependencies : deps_tools,
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -413,6 +418,7 @@
 	   libinput_measure_touchpad_tap_sources,
 	   dependencies : deps_tools,
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -438,6 +444,7 @@
 		   debug_gui_sources,
 		   dependencies : deps_debug_gui,
 		   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 		   install_dir : libinput_tool_path,
 		   install : true
 		   )
@@ -455,6 +462,7 @@
 	   libinput_sources,
 	   dependencies : deps_tools,
 	   include_directories : include_directories ('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true
 	   )
 configure_file(input : 'tools/libinput.man',
@@ -471,6 +479,7 @@
 	   ptraccel_debug_sources,
 	   dependencies : [ dep_libfilter, dep_libinput ],
 	   include_directories : include_directories('src'),
+	   install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : false
 	   )
 
