--- meson.build.orig	2018-07-23 21:31:28.741091000 +0200
+++ meson.build	2018-07-23 21:38:11.131091000 +0200
@@ -107,12 +107,14 @@
 	   'udev/libinput-device-group.c',
 	   dependencies : [dep_udev, dep_libwacom],
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true,
 	   install_dir : udev_dir)
 executable('libinput-model-quirks',
 	   'udev/libinput-model-quirks.c',
 	   dependencies : [dep_udev, dep_libevdev],
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true,
 	   install_dir : udev_dir)
 
@@ -425,6 +427,7 @@
 	   libinput_debug_events_sources,
 	   dependencies : deps_tools,
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true
 	   )
@@ -440,6 +443,7 @@
 	   libinput_list_devices_sources,
 	   dependencies : deps_tools,
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -455,6 +459,7 @@
 	   libinput_measure_sources,
 	   dependencies : deps_tools,
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -514,6 +519,7 @@
 	   libinput_record_sources,
 	   dependencies : deps_tools + [dep_udev],
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install_dir : libinput_tool_path,
 	   install : true,
 	   )
@@ -548,6 +554,7 @@
 		   debug_gui_sources,
 		   dependencies : deps_debug_gui,
 		   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 		   install_dir : libinput_tool_path,
 		   install : true
 		   )
@@ -565,6 +572,7 @@
 	   libinput_sources,
 	   dependencies : deps_tools,
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : true
 	   )
 configure_file(input : 'tools/libinput.man',
@@ -579,6 +587,7 @@
 	   ptraccel_debug_sources,
 	   dependencies : [ dep_libfilter, dep_libinput ],
 	   include_directories : [includes_src, includes_include],
+       install_rpath : '@PREFIX@/lib/x86_64-linux-gnu',
 	   install : false
 	   )
 
