diff --git a/meson.build b/meson.build
index 9fc09a4..bfdbaad 100644
--- a/meson.build
+++ b/meson.build
@@ -426,7 +426,7 @@ elif _egl == 'enabled'
     error('EGL requires shared-glapi')
   elif not ['disabled', 'dri'].contains(with_glx)
     error('EGL requires dri, but a GLX is being built without dri')
-  elif ['darwin', 'windows'].contains(host_machine.system())
+  elif ['windows'].contains(host_machine.system())
     error('EGL is not available on Windows or MacOS')
   endif
   with_egl = true
@@ -870,16 +870,16 @@ if with_platform_haiku
 endif
 
 prog_python = import('python').find_installation('python3')
-has_mako = run_command(
-  prog_python, '-c',
-  '''
-from distutils.version import StrictVersion
-import mako
-assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
-  ''')
-if has_mako.returncode() != 0
-  error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
-endif
+#has_mako = run_command(
+  #prog_python, '-c',
+  #'''
+#from distutils.version import StrictVersion
+#import mako
+#assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
+  #''')
+#if has_mako.returncode() != 0
+  #error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
+#endif
 
 if cc.get_id() == 'gcc' and cc.version().version_compare('< 4.4.6')
   error('When using GCC, version 4.4.6 or later is required.')
@@ -1324,11 +1324,11 @@ elif with_dri_i965 and with_shader_cache
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
+#if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
   dep_clock = null_dep
-else
-  dep_clock = cc.find_library('rt')
-endif
+#else
+  #dep_clock = cc.find_library('rt')
+#endif
 
 # TODO: some of these may be conditional
 dep_zlib = dependency('zlib', version : '>= 1.2.3', fallback : ['zlib', 'zlib_dep'])
