diff --git a/orig.meson.build b/meson.build
index 07c77e1..a66cfb3 100644
--- a/orig.meson.build
+++ b/meson.build
@@ -480,8 +480,6 @@ elif _egl == 'enabled'
     error('EGL requires shared-glapi')
   elif not ['disabled', 'dri'].contains(with_glx)
     error('EGL requires dri, but a GLX is being built without dri')
-  elif host_machine.system() == 'darwin'
-    error('EGL is not available on MacOS')
   endif
   with_egl = true
 else
@@ -1510,11 +1508,11 @@ elif with_dri_i965 and with_shader_cache
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
+#if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
   dep_clock = null_dep
-else
-  dep_clock = cc.find_library('rt')
-endif
+#else
+  #dep_clock = cc.find_library('rt')
+#endif
 
 dep_zlib = dependency('zlib', version : '>= 1.2.3',
                       fallback : ['zlib', 'zlib_dep'],
