diff --git b/src/loader/orig.loader.c a/src/loader/loader.c
index d64bc7c..90850b2 100644
--- b/src/loader/orig.loader.c
+++ a/src/loader/loader.c
@@ -84,17 +84,19 @@ loader_open_device(const char *device_name)
 {
    int fd;
 #ifdef O_CLOEXEC
+   errno = 0;
    fd = open(device_name, O_RDWR | O_CLOEXEC);
    if (fd == -1 && errno == EINVAL)
 #endif
    {
+      errno = 0;
       fd = open(device_name, O_RDWR);
       if (fd != -1)
          fcntl(fd, F_SETFD, fcntl(fd, F_GETFD) | FD_CLOEXEC);
    }
    if (fd == -1 && errno == EACCES) {
-      log_(_LOADER_WARNING, "failed to open %s: %s\n",
-           device_name, strerror(errno));
+      log_(_LOADER_WARNING, "loader_open_device() pid=%d failed to open %s: %s\n",
+           getpid(), device_name, strerror(errno));
    }
    return fd;
 }
