diff --git a/src/core/load-fragment.c b/src/core/load-fragment.c
index 0baf08e..37b072f 100644
--- a/src/core/load-fragment.c
+++ b/src/core/load-fragment.c
@@ -5,6 +5,8 @@
 
 #include <errno.h>
 #include <fcntl.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 #include <linux/oom.h>
 #include <sched.h>
diff --git a/src/import/import-raw.c b/src/import/import-raw.c
index f7ed163..f2fcb59 100644
--- a/src/import/import-raw.c
+++ b/src/import/import-raw.c
@@ -1,5 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 
 #include "sd-daemon.h"
diff --git a/src/import/import-tar.c b/src/import/import-tar.c
index 9020270..66b82d6 100644
--- a/src/import/import-tar.c
+++ b/src/import/import-tar.c
@@ -1,5 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 
 #include "sd-daemon.h"
diff --git a/src/import/pull-raw.c b/src/import/pull-raw.c
index 66c3f65..cd535ce 100644
--- a/src/import/pull-raw.c
+++ b/src/import/pull-raw.c
@@ -1,6 +1,8 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <curl/curl.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 #include <sys/xattr.h>
 
diff --git a/src/journal/journalctl.c b/src/journal/journalctl.c
index 7f3dcd5..6a843de 100644
--- a/src/journal/journalctl.c
+++ b/src/journal/journalctl.c
@@ -4,6 +4,8 @@
 #include <fcntl.h>
 #include <fnmatch.h>
 #include <getopt.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 #include <signal.h>
 #include <stddef.h>
diff --git a/src/libsystemd/sd-journal/journal-file.c b/src/libsystemd/sd-journal/journal-file.c
index d2493a0..b7825f4 100644
--- a/src/libsystemd/sd-journal/journal-file.c
+++ b/src/libsystemd/sd-journal/journal-file.c
@@ -2,6 +2,8 @@
 
 #include <errno.h>
 #include <fcntl.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 #include <linux/magic.h>
 #include <pthread.h>
diff --git a/src/shared/btrfs-util.c b/src/shared/btrfs-util.c
index b3e4b50..d6c2400 100644
--- a/src/shared/btrfs-util.c
+++ b/src/shared/btrfs-util.c
@@ -3,6 +3,8 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <inttypes.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/btrfs_tree.h>
 #include <linux/fs.h>
 #include <linux/loop.h>
diff --git a/src/tmpfiles/tmpfiles.c b/src/tmpfiles/tmpfiles.c
index bc83aab..311ce8c 100644
--- a/src/tmpfiles/tmpfiles.c
+++ b/src/tmpfiles/tmpfiles.c
@@ -5,6 +5,8 @@
 #include <fnmatch.h>
 #include <getopt.h>
 #include <limits.h>
+// RJVB
+#include <sys/mount.h>
 #include <linux/fs.h>
 #include <stdbool.h>
 #include <stddef.h>
