diff --git a/meson.build.orig b/meson.build
index 8963b31..d50b43a 100644
--- a/meson.build.orig
+++ b/meson.build
@@ -444,7 +444,6 @@ with_xlib_lease = get_option('xlib-lease') \
   .allowed()
 
 with_egl = get_option('egl') \
-  .require(host_machine.system() != 'darwin', error_message : 'EGL not supported on MacOS') \
   .require(with_platform_windows or with_platform_haiku or with_dri or with_platform_android, error_message : 'EGL requires DRI, Haiku, Windows or Android') \
   .require(with_shared_glapi, error_message : 'EGL requires shared-glapi') \
   .require(with_glx != 'xlib', error_message :'EGL requires DRI, but GLX is being built with xlib support') \
@@ -1528,11 +1527,11 @@ if with_gallium_i915 and host_machine.cpu_family().startswith('x86') == false
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
+#if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
   dep_clock = null_dep
-else
-  dep_clock = cc.find_library('rt')
-endif
+#else
+  #dep_clock = cc.find_library('rt')
+#endif
 
 dep_zlib = dependency('zlib', version : '>= 1.2.3',
                       allow_fallback: true,
