--- mesa-18.3.3/src/intel/vulkan/anv_allocator.c	2019-02-10 14:59:36.075814345 +0100
+++ mesa-18.3.3/src/intel/vulkan/orig.anv_allocator.c	2019-01-31 22:05:07.000000000 +0100
@@ -110,12 +110,6 @@
 #define ANV_MMAP_CLEANUP_INIT ((struct anv_mmap_cleanup){0})
 
 #ifndef HAVE_MEMFD_CREATE
-
-#ifndef SYS_memfd_create
-#define __NR_memfd_create 279
-#define SYS_memfd_create __NR_memfd_create 
-#endif
-
 static inline int
 memfd_create(const char *name, unsigned int flags)
 {
--- mesa-18.3.3/src/intel/vulkan/anv_gem_stubs.c	2019-02-10 15:21:10.495814345 +0100
+++ mesa-18.3.3/src/intel/vulkan/orig.anv_gem_stubs.c	2019-01-31 22:05:07.000000000 +0100
@@ -28,11 +28,6 @@
 #include "anv_private.h"
 
 #ifndef HAVE_MEMFD_CREATE
-#ifndef SYS_memfd_create
-#define __NR_memfd_create 279
-#define SYS_memfd_create __NR_memfd_create 
-#endif
-
 static inline int
 memfd_create(const char *name, unsigned int flags)
 {
