diff --git Lib/test/test_sqlite3/test_dbapi.py.orig Lib/test/test_sqlite3/test_dbapi.py
index 7ad4136..2fd925b 100644
--- Lib/test/test_sqlite3/test_dbapi.py.orig
+++ Lib/test/test_sqlite3/test_dbapi.py
@@ -679,24 +679,24 @@ class OpenTests(unittest.TestCase):
             self.assertTrue(os.path.exists(path))
             cx.execute(self._sql)
 
-    def get_undecodable_path(self):
-        path = TESTFN_UNDECODABLE
-        if not path:
-            self.skipTest("only works if there are undecodable paths")
-        try:
-            open(path, 'wb').close()
-        except OSError:
-            self.skipTest(f"can't create file with undecodable path {path!r}")
-        unlink(path)
-        return path
-
-    @unittest.skipIf(sys.platform == "win32", "skipped on Windows")
-    def test_open_with_undecodable_path(self):
-        path = self.get_undecodable_path()
-        self.addCleanup(unlink, path)
-        with contextlib.closing(sqlite.connect(path)) as cx:
-            self.assertTrue(os.path.exists(path))
-            cx.execute(self._sql)
+#     def get_undecodable_path(self):
+#         path = TESTFN_UNDECODABLE
+#         if not path:
+#             self.skipTest("only works if there are undecodable paths")
+#         try:
+#             open(path, 'wb').close()
+#         except OSError:
+#             self.skipTest(f"can't create file with undecodable path {path!r}")
+#         unlink(path)
+#         return path
+# 
+#     @unittest.skipIf(sys.platform == "win32", "skipped on Windows")
+#     def test_open_with_undecodable_path(self):
+#         path = self.get_undecodable_path()
+#         self.addCleanup(unlink, path)
+#         with contextlib.closing(sqlite.connect(path)) as cx:
+#             self.assertTrue(os.path.exists(path))
+#             cx.execute(self._sql)
 
     def test_open_uri(self):
         path = TESTFN
@@ -732,14 +732,14 @@ class OpenTests(unittest.TestCase):
             with self.assertRaises(sqlite.OperationalError):
                 cx.execute(self._sql)
 
-    @unittest.skipIf(sys.platform == "win32", "skipped on Windows")
-    def test_open_undecodable_uri(self):
-        path = self.get_undecodable_path()
-        self.addCleanup(unlink, path)
-        uri = "file:" + urllib.parse.quote(path)
-        with contextlib.closing(sqlite.connect(uri, uri=True)) as cx:
-            self.assertTrue(os.path.exists(path))
-            cx.execute(self._sql)
+#     @unittest.skipIf(sys.platform == "win32", "skipped on Windows")
+#     def test_open_undecodable_uri(self):
+#         path = self.get_undecodable_path()
+#         self.addCleanup(unlink, path)
+#         uri = "file:" + urllib.parse.quote(path)
+#         with contextlib.closing(sqlite.connect(uri, uri=True)) as cx:
+#             self.assertTrue(os.path.exists(path))
+#             cx.execute(self._sql)
 
     def test_factory_database_arg(self):
         def factory(database, *args, **kwargs):
