diff --git plugins/housekeeping/orig.gsd-disk-space.c plugins/housekeeping/gsd-disk-space.c
index e6eef0f..c998ba2 100644
--- plugins/housekeeping/orig.gsd-disk-space.c
+++ plugins/housekeeping/gsd-disk-space.c
@@ -1017,7 +1017,9 @@ gsd_ldsm_clean (void)
         g_clear_object (&ldsm_monitor);
         g_clear_object (&settings);
         g_clear_object (&privacy_settings);
-        g_clear_pointer (&notification, notify_notification_close);
+        /* NotifyNotification::closed callback will drop reference */
+        if (notification != NULL)
+                notify_notification_close (notification, NULL);
         g_slist_free_full (ignore_paths, g_free);
         ignore_paths = NULL;
 }
