diff --git modules/gui/qt/components/interface_widgets.cpp.orig modules/gui/qt/components/interface_widgets.cpp
index cb48dc5..0f699a0 100644
--- modules/gui/qt/components/interface_widgets.cpp.orig
+++ modules/gui/qt/components/interface_widgets.cpp
@@ -52,6 +52,15 @@ static inline int qrand() {
 }
 #endif
 
+// RJVB
+#if defined(QT5_HAS_XCB)
+# include <xcb/xcb.h>
+#endif
+#if defined (QT5_HAS_X11)
+# include <X11/Xlib.h>
+# include <QX11Info>
+#endif
+
 #if defined (QT_HAS_X11)
 # include <X11/Xlib.h>
 # include "qt_x11.hpp"
