diff --git modules/codec/fluidsynth.c.orig modules/codec/fluidsynth.c
index c68c222..e94ccd3 100644
--- modules/codec/fluidsynth.c.orig
+++ modules/codec/fluidsynth.c
@@ -129,8 +129,8 @@ static int Open (vlc_object_t *p_this)
     {
         glob_t gl;
 
-        glob("/usr/share/sounds/sf2/*.sf2", GLOB_NOESCAPE, NULL, &gl);
-        glob("/usr/share/soundfonts/*.sf2", GLOB_NOESCAPE | GLOB_APPEND, NULL,
+        glob("@PREFIX@/share/sounds/sf2/*.sf2", GLOB_NOESCAPE, NULL, &gl);
+        glob("@PREFIX@/share/soundfonts/*.sf2", GLOB_NOESCAPE | GLOB_APPEND, NULL,
              &gl);
 
         for (size_t i = 0; i < gl.gl_pathc; i++)
