--- cmake/orig.FindIconv.cmake	2017-10-01 16:02:02.654218696 +0200
+++ cmake/FindIconv.cmake	2017-10-01 16:02:08.658981870 +0200
@@ -41,7 +41,7 @@
 
 if(NOT ICONV_INCLUDE_DIR STREQUAL "ICONV_INCLUDE_DIR-NOTFOUND")
     set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
-    check_function_exists(iconv_open ICONV_IN_GLIBC)
+    #check_function_exists(iconv_open ICONV_IN_GLIBC)
 endif()
 
 if(NOT ICONV_IN_GLIBC)
@@ -69,6 +69,8 @@
     set(ICONV_TEST "In glibc")
 endif()
 
+message(STATUS "ICONV_TEST=${ICONV_TEST}")
+
 set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
 set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARY})
 
