diff --git src/shortcuts/dzl-shortcut-theme-editor.c src/shortcuts/dzl-shortcut-theme-editor.c
index 94abbd9..82faf57 100644
--- src/shortcuts/dzl-shortcut-theme-editor.c
+++ src/shortcuts/dzl-shortcut-theme-editor.c
@@ -199,7 +199,7 @@ dzl_shortcut_theme_editor_row_activated (DzlShortcutThemeEditor *self,
                              "shortcut-title", title,
                              "title", _("Set Shortcut"),
                              "transient-for", toplevel,
-                             "use-header-bar", TRUE,
+                             "use-header-bar", FALSE,
                              NULL);
 
       g_signal_connect_object (dialog,
diff --git src/shortcuts/dzl-shortcuts-window.c src/shortcuts/dzl-shortcuts-window.c
index 0b91aa6..466fdf0 100644
--- src/shortcuts/dzl-shortcuts-window.c
+++ src/shortcuts/dzl-shortcuts-window.c
@@ -880,11 +880,11 @@ dzl_shortcuts_window_init (DzlShortcutsWindow *self)
   priv->search_text_group = gtk_size_group_new (GTK_SIZE_GROUP_HORIZONTAL);
   priv->search_image_group = gtk_size_group_new (GTK_SIZE_GROUP_HORIZONTAL);
 
-  priv->header_bar = g_object_new (GTK_TYPE_HEADER_BAR,
-                                   "show-close-button", TRUE,
-                                   "visible", TRUE,
-                                   NULL);
-  gtk_window_set_titlebar (GTK_WINDOW (self), GTK_WIDGET (priv->header_bar));
+//   priv->header_bar = g_object_new (GTK_TYPE_HEADER_BAR,
+//                                    "show-close-button", TRUE,
+//                                    "visible", TRUE,
+//                                    NULL);
+//   gtk_window_set_titlebar (GTK_WINDOW (self), GTK_WIDGET (priv->header_bar));
 
   search_button = g_object_new (GTK_TYPE_TOGGLE_BUTTON,
                                 "child", g_object_new (GTK_TYPE_IMAGE,
@@ -894,7 +894,7 @@ dzl_shortcuts_window_init (DzlShortcutsWindow *self)
                                 "visible", TRUE,
                                 NULL);
   gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (search_button)), "image-button");
-  gtk_container_add (GTK_CONTAINER (priv->header_bar), GTK_WIDGET (search_button));
+//   gtk_container_add (GTK_CONTAINER (priv->header_bar), GTK_WIDGET (search_button));
 
   priv->main_box = g_object_new (GTK_TYPE_BOX,
                            "orientation", GTK_ORIENTATION_VERTICAL,
@@ -921,7 +921,7 @@ dzl_shortcuts_window_init (DzlShortcutsWindow *self)
   priv->title_stack = g_object_new (GTK_TYPE_STACK,
                                     "visible", TRUE,
                                     NULL);
-  gtk_header_bar_set_custom_title (priv->header_bar, GTK_WIDGET (priv->title_stack));
+//   gtk_header_bar_set_custom_title (priv->header_bar, GTK_WIDGET (priv->title_stack));
 
   label = gtk_label_new (_("Shortcuts"));
   gtk_widget_show (label);
